pallets/utxo/src/lib.rs

```diff
@@ -841,10 +820,29 @@ pub mod pallet {
             value: Value,
             address: Vec<u8>,
         ) -> DispatchResultWithPostInfo {
-            use chainscript::Script;
+            let (_, data, _) = bech32::decode(&address).map_err(|e| match e {
+                bech32::Error::InvalidLength => {
+                    DispatchError::Other("Failed to decode address: invalid length")
+                }
+                bech32::Error::InvalidChar(_) => {
+                    DispatchError::Other("Failed to decode address: invalid character")
+                }
+                bech32::Error::MixedCase => {
+                    DispatchError::Other("Failed to decode address: mixed case")
+                }
+                bech32::Error::InvalidChecksum => {
+                    DispatchError::Other("Failed to decode address: invalid checksum")
+                }
+                bech32::Error::InvalidHrp => {
+                    DispatchError::Other("Failed to decode address: invalid HRP")
+                }
+                _ => DispatchError::Other("Failed to decode address"),
+            })?;
 
+            let mut slice: &[u8] = &data;
+            let dest: Destination<T::AccountId> = Destination::decode(&mut slice)
```

Comment:
This is kind of ugly, does anyone know if there's a simpler way of converting `Vec<u8>` to a `&[u8]` without the compiler complaining about size not being known at compile time

Reply:
`decode fn` requires implementation of `Input` trait for those types that might be decoded. Unfortunately, it does not implement for each case.